llvm/unittests/Transforms/Vectorize/SandboxVectorizer/SeedCollectorTest.cpp

```diff
@@ -0,0 +1,101 @@
+//===- SeedCollectorTest.cpp ----------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "llvm/Transforms/Vectorize/SandboxVectorizer/SeedCollector.h"
+#include "llvm/AsmParser/Parser.h"
+#include "llvm/SandboxIR/Function.h"
+#include "llvm/SandboxIR/Instruction.h"
+#include "llvm/Support/SourceMgr.h"
+#include "gtest/gtest.h"
+
+using namespace llvm;
+
+struct SeedBundleTest : public testing::Test {
+  LLVMContext C;
+  std::unique_ptr<Module> M;
+
+  void parseIR(LLVMContext &C, const char *IR) {
+    SMDiagnostic Err;
+    M = parseAssemblyString(IR, Err, C);
+    if (!M)
+      Err.print("LegalityTest", errs());
+  }
+};
+
+TEST_F(SeedBundleTest, SeedBundle) {
+  parseIR(C, R"IR(
+define void @foo(float %v0, float %v1) {
+bb:
+  %add0 = fadd float %v0, %v1
+  %add1 = fadd float %v0, %v1
+  %add2 = fadd float %v0, %v1
+  %add3 = fadd float %v0, %v1
+  %add4 = fadd float %v0, %v1
+  ret void
+}
+)IR");
+  Function &LLVMF = *M->getFunction("foo");
+  sandboxir::Context Ctx(C);
+  auto &F = *Ctx.createFunction(&LLVMF);
+  DataLayout DL(M->getDataLayout());
+  auto *BB = &*F.begin();
+  auto It = BB->begin();
+  auto *I0 = &*It++;
+  auto *I1 = &*It++;
+  // Assume test instructions are identical in the number of bits.
+  const unsigned kFloatBits = sandboxir::Utils::getNumBits(I0, DL);
```

Comment:
nit: Not sure about the small case `k` isn't the linter complaining about it?
Btw I don't understand why you named it `kFloatBits`, what does `k` refer to ? I would have expected something like `I0Bits`.

Reply:
Linter doesn't complain. kFloatbits because the type is a constant float, and that is somewhat more relevant when figuring out if the test is getting the right number of bits. But all updated.